src/main/scala/net/ripe/rpki/publicationserver/repository/DataFlusher.scala

```diff
@@ -147,11 +148,16 @@ class DataFlusher(conf: AppConfig)(implicit val system: ActorSystem)
     }
 
     val deltas = pgStore.getReasonableDeltas(sessionId)
-
-    deltas.filter(_._1 != latestSerial).foreach { case (serial, _) =>
-      withAtomicStream(deltaPath(sessionId, serial), rrdpWriter.fileAttributes) {
-        writeRrdpDelta(sessionId, serial, _)
-      }
+    for {
+      (serial, _) <- deltas
+      // Delta for the latest serial was already created above, so we can skip it here.
+      if serial != latestSerial
+    } {
+      val (deltaHash, deltaSize) =
+        withAtomicStream(deltaPath(sessionId, serial), rrdpWriter.fileAttributes) {
+          writeRrdpDelta(sessionId, serial, _)
+        }
+      pgStore.updateDeltaInfo(sessionId, serial, deltaHash, deltaSize)
```

Comment:
Is it correct that we do `updateDeltaInfo` for all deltas except for the last one?
Shouldn't it be the opposite?

Reply:
We did (and do) update the info for the latest serial as well: https://github.com/RIPE-NCC/rpki-publication-server/pull/38/files#diff-37e6a0a4ebc4a9797056afc8b8daa5f5e3072ba6664ac8f763e629a7b6fbda65R147, but that part of the code is hidden by the default GitHub diff view, since it has not changed.